homeassistant/helpers/entity.py

```diff
@@ -319,7 +319,14 @@ def _async_write_ha_state(self) -> None:
             sstate = self.state
             state = STATE_UNKNOWN if sstate is None else str(sstate)
             attr.update(self.state_attributes or {})
-            attr.update(self.device_state_attributes or {})
+            entity_state_attributes = self.entity_state_attributes
+            # Backwards compatibility for "device_state_attributes" deprecated in 2021.4
+            # Add warning in 2021.6, remove in 2021.10
+            if entity_state_attributes is None and hasattr(
```

Comment:
Maybe use `getattr` (with default `None`) here already and a walrus operator?

Reply:
Updated to keep `Entity.device_state_attributes` during deprecation period instead.